src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1378 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.SplitOrMergeIfStatements;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SplitOrMergeIfStatements
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsMergeConsecutiveIfStatements)]
+    public sealed class MergeConsecutiveIfStatementsTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpMergeConsecutiveIfStatementsCodeRefactoringProvider();
+
+        [Theory]
+        [InlineData("[||]else if (b)")]
+        [InlineData("el[||]se if (b)")]
+        [InlineData("else[||] if (b)")]
+        [InlineData("else [||]if (b)")]
+        [InlineData("else i[||]f (b)")]
+        [InlineData("else if[||] (b)")]
+        [InlineData("else if [||](b)")]
+        [InlineData("else if (b)[||]")]
+        [InlineData("else [|if|] (b)")]
+        [InlineData("else [|if (b)|]")]
+        [InlineData("[|else if (b)|]")]
```

Comment:
this is   a case where a theory works well for me.

Reply:
I hope so. You actually requested I use a theory here, so that's what I did.